### DISCUSSION TITLE
What is the recommended way to store coupons?

### DISCUSSION THREAD

**Discussion by abublihi** [Safemood/discountify#30] [Q&A]
@Safemood Thank you for the awesome package, 

After I read all the docs, the package looks awesome, but I would ask what is the recommended way to store discount coupones?

Because for example the singleUse coupons should stored in some way to check whether the were used or not!

**Comment by Safemood** [comment]
Hi @abublihi 

Thank you for your interest in the package! Sorry for my late response. I had a busy weekend! 🏃‍♂️😊 Currently, Discountify only supports in-memory storage for discounts. I understand your concern regarding the tracking of single-use coupons, and I recognize that this could pose a problem. I see this as a bug and will prioritize working on a fix asap.

**Reply by abublihi** [reply]
Thank you for responding @Safemood ,

وبارك الله فيك

**Reply by Safemood** [reply]
Hi @abublihi 

Now discountify can hold state between requests so we can keep track of single and limited use coupons.

Hope you did check this PR : https://github.com/Safemood/discountify/pull/32

its also merged into the 1.5.1 version.
